host/tests/dsHdmiIn_L3_Tests/dsHdmiIn_test1_ConnectionCallback_Verify.py

```diff
@@ -0,0 +1,184 @@
+            None.
+        """
+
+        self.deviceDownloadPath = self.cpe.get("target_directory")
+
+        #download test artifacts to device
+        url = self.testSetup.assets.device.test1_ConnectionCallback_Verify.artifacts
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+
+    def testCleanAssets(self):
+        """
+        Removes the assets copied to the dut.
+
+        Args:
+            None.
+        """
+        self.deleteFromDevice(self.testStreams)
+
+    def testRunPrerequisites(self):
+        """
+        Runs Prerequisite commands listed in test-setup configuration file on the dut.
+
+        Args:
+            None.
+        """
+
+        #Run test specific commands
+        cmds = self.testSetup.assets.device.test1_ConnectionCallback_Verify.execute
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+    def testUnplugHDMI(self, unplug:True, manual=False):
+        """
+        Verifies whether the Connection status of HdmiIn ports
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+        Returns:
+            bool
+        """
+        if manual == True:
+            if unplug == True:
+                return self.testUserResponse.getUserYN("UnPlug the HDMI Cable? (Y/N):")
+            else :
+                return self.testUserResponse.getUserYN("Plug the HDMI Cable? (Y/N):")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def find_Connectionstatus(self, input_str: str, status: str) -> bool:
+        if status in input_str:
+            return True
+        return False
+
+    def testFunction(self):
+        """This Callback will check for Connect Disconnect events on HdmiIn ports
+
+        Returns:
+            bool
+        """
+
+        # Download the assets listed in test setup configuration file
+        self.testDownloadAssets()
+
+        # Run Prerequisites listed in the test setup configuration file
+        self.testRunPrerequisites()
+
+        # Create the dsHdmiIn class
+        self.testdsHdmiIn = dsHdmiInClass(self.deviceProfile, self.hal_session)
+
+        self.log.testStart("test1_ConnectionCallback_Verify", '1')
+
+        # Initialize the dsAudio module
+        self.testdsHdmiIn.initialise(self.testdsHdmiIn.getDeviceType())
+   
+        # Loop through the supported HdmiIn ports
+        for port,index in self.testdsAudio.getSupportedPorts():
+            self.log.stepStart(f'Select {port} Port')
+            self.log.step(f'Select {port} Port')
+
+            # Plug In HdmiIn device
+            result = self.testUnplugHDMI(True,True)
+
+            result = self.hal_session.read_until("Received Connection status callback port: , Connection:")
+            print(result)
```

Comment:
remove the print, if you want print any message use log module

Reply:
done